### PR TITLE
Complete HW2(2/3)

### PR DESCRIPTION
**Pull request by DmitriySmertin**
Hello. Please check my HW2. Thx for the feedback.

### CODE REVIEW

File: src/test/java/com/epam/tc/hw2/GeneralData.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.epam.tc.hw2;
+
+public class GeneralData {
+    protected  String indexUrl = "https://jdi-testing.github.io/jdi-light/index.html";
+    protected  String expectedBrowserTitle = "Home Page";
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/GeneralData.java]: Why did you add extra space?

File: pom.xml [hunk 2]
```diff
@@ -84,5 +92,17 @@
             <version>1.0</version>
             <systemPath>${project.basedir}/src/main/resources/calculator-1.0.jar</systemPath>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/io.github.bonigarcia/webdrivermanager -->
```
Comment (komyagin) [inline pom.xml]: Why did you add this comment here?

File: pom.xml [hunk 3]
```diff
@@ -84,5 +92,17 @@
...
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>4.4.3</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java -->
```
Comment (komyagin) [inline pom.xml]: Why did you add this comment here?

File: pom.xml [hunk 4]
```diff
@@ -84,5 +92,17 @@
...
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>3.141.59</version>
```
Comment (komyagin) [inline pom.xml]: You should move versions to the properties

File: pom.xml [hunk 5]
```diff
@@ -84,5 +92,17 @@
             <version>1.0</version>
             <systemPath>${project.basedir}/src/main/resources/calculator-1.0.jar</systemPath>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/io.github.bonigarcia/webdrivermanager -->
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>4.4.3</version>
```
Comment (komyagin) [inline pom.xml]: You should move versions to the properties

File: src/test/java/com/epam/tc/hw2/GeneralData.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    protected String userName = "Roman";
+    protected String userPass = "Jdi1234";
+    protected String expectedUserLogin = "ROMAN IOVLEV";
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/GeneralData.java]: And how do declare string constant properly?

File: src/test/java/com/epam/tc/hw2/ex1/Exercise1.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+import org.testng.annotations.Test;
+
+
+public class Exercise1 extends SetUpAndTearDown {
+
+    //Data
+    private String homeNavItem = "HOME";
+    private String contactNavItem = "CONTACT FORM";
+    private String serviceNavItem = "SERVICE";
+    private String metalsAndColorsNavItem = "METALS & COLORS";
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/ex1/Exercise1.java]: And how do declare string constant properly?

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw2/GeneralData.java (hunk 1 above)
Comment (DmitriySmertin) [reply src/test/java/com/epam/tc/hw2/GeneralData.java]: Hmm, to be honest, I didn't think about allocating space. Tell me, should I make them private? And getter methods. Or make it public?

File: pom.xml (hunk 2 above)
Comment (DmitriySmertin) [reply pom.xml]: I copied it from the webdriver manager documentation.

File: pom.xml (hunk 3 above)
Comment (DmitriySmertin) [reply pom.xml]: I copied it from the selenium documentation.

File: pom.xml (hunk 4 above)
Comment (DmitriySmertin) [reply pom.xml]: Understood

File: src/test/java/com/epam/tc/hw2/ex1/Exercise1.java (hunk 7 above)
Comment (DmitriySmertin) [reply src/test/java/com/epam/tc/hw2/ex1/Exercise1.java]: Maybe private final

File: src/test/java/com/epam/tc/hw2/GeneralData.java (hunk 6 above)
Comment (DmitriySmertin) [reply src/test/java/com/epam/tc/hw2/GeneralData.java]: Maybe private final

File: pom.xml (hunk 5 above)
Comment (DmitriySmertin) [reply pom.xml]: Understood

**Comment by DmitriySmertin** [comment]
Fixed (2/3)

File: src/test/java/com/epam/tc/hw2/GeneralData.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package com.epam.tc.hw2;
+
+public class GeneralData {
+
+    private final String indexUrl = "https://jdi-testing.github.io/jdi-light/index.html";
+    private final String expectedBrowserTitle = "Home Page";
+    private final String userName = "Roman";
+    private final String userPass = "Jdi1234";
+    private final String expectedUserLogin = "ROMAN IOVLEV";
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/GeneralData.java]: Due to convention you should use UPPER_CASE for the names of the constants

File: src/test/java/com/epam/tc/hw2/ex1/Exercise1.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+public class Exercise1 extends SetUpAndTearDown {
+
+    //Data
+    GeneralData data = new GeneralData();
+    private final String homeNavItem = "HOME";
+    private final String contactNavItem = "CONTACT FORM";
+    private final String serviceNavItem = "SERVICE";
+    private final String metalsAndColorsNavItem = "METALS & COLORS";
+    private final List<String> expectedTextBenefitList = Arrays.asList(
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/ex1/Exercise1.java]: The same

File: src/test/java/com/epam/tc/hw2/ex1/Exercise1.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        //8. Assert that there is the iframe with “Frame Button” exist
+        Assertions.assertThat(driver.findElement(By.id("frame")).isEnabled());
+        //9.  Switch to the iframe and check that there is “Frame Button” in the iframe
+        String windowHomePage = driver.getWindowHandle();
+        driver.switchTo().frame("frame");
+        Assertions.assertThat(driver.findElement(By.id("frame-button")).isEnabled());
+        //10. Switch to original window back
+        driver.switchTo().window(windowHomePage);
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        List<WebElement> leftMenuItemsList = driver.findElements(By.xpath("//ul[contains (@class, 'left')]/li/a"));
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/ex1/Exercise1.java]: There is more suitable locator

File: src/test/java/com/epam/tc/hw2/ex1/Exercise1.java [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+import org.assertj.core.api.Assertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+
+public class Exercise1 extends SetUpAndTearDown {
+
+    //Data
+    GeneralData data = new GeneralData();
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/ex1/Exercise1.java]: Are you sure about it?

File: src/test/java/com/epam/tc/hw2/ex2/Exercise2.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+import com.epam.tc.hw2.SetUpAndTearDown;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+public class Exercise2 extends SetUpAndTearDown {
+
+    GeneralData data = new GeneralData();
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw2/ex2/Exercise2.java]: Do you really need this?

**Review by komyagin — CHANGES_REQUESTED** [review]

**Review by komyagin — APPROVED** [review]

**Comment by DmitriySmertin** [comment]
merge
